docs/source/API/core/memory_spaces.md

```diff
@@ -35,6 +35,18 @@
 
 `Kokkos::HostSpace` is a [`MemorySpace` type](MemorySpaceConcept) representing traditional random access memory accessible from the CPU.  Except in rare instances, it should not be used directly, but instead should be used generically as an memory space.  For details, see [the documentation on the `MemorySpace` concept](MemorySpaceConcept).
 
+(SharedSpace)=
+## `Kokkos::SharedSpace`
+
+`Kokkos::SharedSpace` <sup>since 4.0</sup> is a [`MemorySpace` type](MemorySpaceConcept) alias representing memory that can be accessed by any enabled [`ExecutionSpace` type](ExecutionSpaceConcept). To achieve this, the memory can be moved to and from the local memory of the processing units represented by the `ExecutionSpaces`. The movement is done automatically by the os and driver at the moment of access. If not currently located in the local memory of the accessing processing unit, the memory is moved in chunks (size is backend dependent). These chunks can be moved independently (e.g. only the part that is accessed on the GPU is moved to the GPU) and are treated like local memory while residing on the processing unit. For details, see [the documentation on the `MemorySpace` concept](MemorySpaceConcept).
```

Comment:
```suggestion
`Kokkos::SharedSpace` <sup>since 4.0</sup> is a [`MemorySpace` type](MemorySpaceConcept) alias representing memory that can be accessed by any enabled [`ExecutionSpace` type](ExecutionSpaceConcept). To achieve this, the memory can be moved to and from the local memory of the processing units represented by the `ExecutionSpaces`. The movement is done automatically by the OS and driver at the moment of access. If not currently located in the local memory of the accessing processing unit, the memory is moved in chunks (size is backend dependent). These chunks can be moved independently (e.g. only the part that is accessed on the GPU is moved to the GPU) and are treated like local memory while residing on the processing unit. For details, see [the documentation on the `MemorySpace` concept](MemorySpaceConcept).
```